sips/mev.md

```diff
@@ -0,0 +1,87 @@
+| Author                    | Title                           | Category | Status              |
+| ------------------------- | ------------------------------- | -------- | ------------------- |
+| Moshe Revah (@moshe-blox) | Support externally built blocks | Core     | open-for-discussion |
+
+## Summary
+
+Support the production and proposal of blocks built by external entities, allowing operators to access blocks from a marketplace of builders such as [Flashbots](https://boost-relay.flashbots.net/).
+
+## Rational & Design Goals
+
+Block building is a computationally intensive task, and even more so when blockspace utilization and profit maximization are in mind. Additionally, many of the more profitable transactions, so-called MEV, are private and only accessible to validators through external block builders.
+
+Currently, operators wishing to offer competitive returns must be sophisticated block builders with private transaction flows. This SIP aims to change that by allowing them to register with and propose blocks from external builders who comply to [ethereum/builder-specs](https://github.com/ethereum/builder-specs).
+
+## Specification
+
+### MEV-supporting operators
+
+Operators who wish to propose blinded blocks must configure their node to do so, otherwise it will propose standard locally-built blocks.
+
+### Validator registration
+
+Builders require validators to publish a [`builder-specs/SignedValidatorRegistration`](https://ethereum.github.io/builder-specs/#model-SignedValidatorRegistration) to set their `fee_recipient` and `gas_limit` preferences.
+
+When building a block for a validator, builders refer only to the registration with the highest timestamp.
+
+In the wild, some Ethereum validator clients currently produce and publish this message every epoch.
+
+#### Fee recipients
+
+Validators may set their preferred `fee_receipient` address by calling `setFeeRecipientAddress` in the `SSVNetwork` contract. Validators may repeat this call as their preference changes over time.
+
+#### Signing
+
+At the start of every slot, operators select their active validators with `ShouldRegisterValidatorAtSlot`, and for those, produce a [`builder-specs/SignedValidatorRegistration`](https://ethereum.github.io/builder-specs/#model-SignedValidatorRegistration) with their preferred `fee_recipient`.
+
+```go
+ValidatorRegistrationSlotInterval = 16 * SlotsPerEpoch
+
+func ShouldRegisterValidatorAtSlot(index phase0.ValidatorIndex, slot phase0.Slot) bool {
+    return (index + slot) % ValidatorRegistrationSlotInterval == 0
```

Comment:
why is validator index important here?

Reply:
That's just used to reduce the CPU/network load by distributing the signing of `ValidatorRegistration` across many slots (10 epochs in the [most recent commit](https://github.com/bloxapp/SIPs/blob/b01c2885c0fd1a9a17e75656ad5f891892166c7b/sips/mev.md#signing)), so each validator would have a registration duty scheduled at a deterministic slot.

We can use the public key instead, which would allow to register non-active validators to MEV builders (although I'm unsure if it's even possible).